src/main/java/ru/sbt/mipt/oop/StandardEventProcessor.java

```diff
@@ -0,0 +1,45 @@
+package ru.sbt.mipt.oop;
+
+import static ru.sbt.mipt.oop.SensorEventType.*;
+
+public class StandardEventProcessor implements EventProcessor{
+
+    public void processEvent(SmartHome smartHome, SensorEvent event) {
+        System.out.println("Got event: " + event);
+        if (event.getType() == LIGHT_ON || event.getType() == LIGHT_OFF) {
+            // событие от источника света
+            Room room = smartHome.getRoomByLight(event.getObjectId());
+            if (room != null) {
+                Light light = room.getLight(event.getObjectId());
+                if (event.getType() == LIGHT_ON) {
+                    Action action = new LightOff(light, room);
+                    action.act();
+                } else {
+                    Action action = new LightOn(light, room);
+                    action.act();
+                }
+            }
+        }
+        if (event.getType() == DOOR_OPEN || event.getType() == DOOR_CLOSED) {
+            // событие от двери
+            Room room = smartHome.getRoomByDoor(event.getObjectId());
+            if (room != null) {
+                Door door = room.getDoor(event.getObjectId());
+                if (event.getType() == DOOR_OPEN) {
+                    Action action = new DoorOpen(door, room);
+                    action.act();
+                } else {
+                    // если мы получили событие о закрытие двери в холле - это значит, что была закрыта входная дверь.
+                    // в этом случае мы хотим автоматически выключить свет во всем доме (это же умный дом!)
+                    if (room.getName().equals("hall")) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0